### PR TITLE
Security Buffs: The PR!

### PR DESCRIPTION
**Pull request by Super3222**
#### Intent of your Pull Request

# - Zoom Code Refactor:

The zoom feature on guns (aka snipers) are now more independent. They reside inside of scopes now, which CANNOT be "attached" to guns.

Guns will initialize with scopes and their action button differently, but won't be any different in gameplay than it was before.

# - Securty Binoculars (**This needs a sprite**):

Out of refactoring the zoom code, I'm implementing security binoculars. They are exactly as they sound. It's like the zoom feature, but with these binoculars you can actually toggle the range of what you want to see.

It can only go from 1 tile - 15 tiles. It's good for scouting areas from a safe position or for looking into places you don't particularly trust.

# - SecHUD Click Code Refactor:

Again, has nothing to do with gameplay. It's just another refactor of my old code.

# - Energy Bolas (**Discontinued**):

Security now has their own brand of bolas!
It's actually pretty unique, check this out...

~~The way it works is by first activating it in your hand, and than~~ throw it at someone. It will entangle their feet, like any other bola, and wait for them to get back up.

If that person ever (willingly) moves, than the bolas will send a shockwave through their body stunning them as well as moving them. It also has the same shock animation that you get from doors, so it's pretty cool.

So yeah it's fun to see a dickhead try to get away, but every step he takes he gets electrocuted. If you're worried about asimov, technically it would be self harm given that they're trapped by the bolas. ~~Yes, it does do a very small amount of burn (2 or 3) outside of being stunned.~~ 0 burn damage.

People can still take it off as usual of course. The stun power will also diminish every time it goes off.

~~Like I said earlier, you have to activate in your hand to use it. This takes security level ID to do. If you don't have security level ID, than it will briefly shock you (with animation).~~

Reblanace Notes:
In the last commit:

Electrical Bolas have been reworked somewhat.
- They no longer start with 6 charges. Instead they start with 1.
- If one is tangled around your leg you have 10 seconds until it begins to shock you after every step you take.
- The charge number will still deplete like normal. Every step you take, there's a shock, and the amount of charges left determines how much you're stunned for and weakened.
- Electrical bolas are no longer ID restricted
- To gain more charges officers will need to "charge" the bolas by activating it in their hands.
- The charges will grow the longer it's in their hands for. The charge number cannot go over 6.
- There is a second timer which is used to kill the charges. This will only happen if an officer takes too long to actually use the bola. If this happens the charge rate will be reduced to 1.
- Bolas are no longer on use. To balance his, when a bola is successfully used it will take an additional 20 seconds (per use) to charge it.

# Other Electric Bolas

Ported from https://github.com/tgstation/tgstation/pull/19006.

# - EMP Grenades:

Honestly I was debating on even doing this, but I've settled that it must be done since it's so minuscule.

The Warden will spawn with a box of EMP grenades inside of his locker. Come on, let's move on from this.

While the EMP rifle is a great weapon you can't look down on this, and tell me that it's bad to have 5 other 1-use EMP items for security to use.

It would also help a LOT with ransacking active clockwork bases if they can get past that stare-y looking piece of shit that fires through walls.

# - Detective Journal (**This needs a sprite**):

I also added a detective journal, which is more or less a photo book.
It also comes with a hand labeler that has twice as many labels as the normal one does.

It's purely for the sake of categorizing crime scenes and all of that.

##### Changelog

:cl:
rscadd: Security Binoculars are at your local secvend! Adjust them between 5-15 tiles.
rscadd: Ported electric bolas from /tg/station. Thank you Papa Bones.
rscadd: The Wardens locker now spawns with a box of EMP grenades. Only 5 of them are in though.
rscadd: The Detective now spawns with a photo book/journal that they can use to categorize cases, store information, and etc. It comes with it's own labeler that has twice as many labels as the normal one does.
/:cl:

### CODE REVIEW

**Comment by Zostroll** [comment]
The energy bolas sound a bit over powered. Could you go into how many sec gets how it is balanced etc?

**Comment by Super3222** [comment]
Sure, Zostroll! Well first, only 10 spawn in the secvendor machine. My original idea was that there'd only be 7 that'd spawn there, but I thought about how every officer would only be carrying them around in pairs. I also considered the fact that they are easily disposable.

Now for the bolas themselves here are some things to consider: They're **all one use**, that means once it entangles someones feet it can't be used again. Even if it doesn't completely deplete. Once it's on someone, the only time that it shocks them is when they attempt to move by striking their arrows keys / WASD. After **five** shocks the bolas will **delete** themselves. They are free to take it off at any time. In fact, as a balancing measure I could make it really easy to take off. There is also no other way to get more bolas unless cargo can order something to fix that, but even that is off on a slim chance. So they are also very scarce in quantity.

The concept behind them was that security is practically going to be stunning the guy they have in these bolas already with a stunbaton or taser gun. Why not make it so that these bolas could be a basis for that? It's also just as hard to get someone in bolas as it is with tasing them. In fact, it'd practically be harder too.

Also an idea for anyone interested in spirting the electrical bolas: They could use a really revealing sprite which doesn't hide the fact that someone has bolas in their hands.

**Comment by Jc1469** [comment]
buffing security doesn't stop security being bad.
it's the players that are bad, not the department's equipment

> So yeah it's fun to see a dickhead try to get away, but every step he takes he gets electrocuted. If you're worried about asimov, technically it would be self harm given that they're trapped by the bolas. Yes, it does do a very small amount of burn (2 or 3) outside of being stunned.

Not true.  Security is harming them by restricting their movement; anyone unsure of how the bolas work is being harmed by security.  So everyone, because we're all idiots here.
May aswell add burn damage to tasers while you're at it.

Security harming people at all, except for the detective who is clinically insane at this point so they get a pass, is bad.  Giving them a standard-issue thing they can get from a vending machine, capable of doing 15 burn damage.  Nice.  Why ever use a taser again?

Be better if you made it every time someone moves at all, meaning if they get dragged, so I can execute people with my 15 damage bolas.

Detective's journal is something nobody will ever use.  Just like the camera, or the evidence bags.  The two things the journal relies on being used.

standard-issue EMP grenades are great.  I love being able to AOE electrify doors randomly, and if the tesla is wired up nice and snug, crit random people for free, as security.

Binoculars sound good though.   Good at doing something the AI could be doing for you but wont be because all of our AI players suck **!!wow!!**


TL;DR 
Bolas are dumb, security gets a license to hurt people, you seem to think AI is fine with that because self harm. Well, AI aint fine with the detective's revolver, so..

Journal wont be used.  Just like the majority of the detective's gear in his office.

EMP grenades are a good addition :+1: I always wanted to have a way to kill the entirety of medbay with one grenade as security, if lucky, of course.

Binoculars are actually a good idea.  It's just funny when you think we have to use binoculars to see past our view range.   Maybe just increase view distance while holding them, with you dropping/putting them away resetting it.

**Comment by Super3222** [comment]
> Bolas are dumb, security gets a license to hurt people, you seem to think AI is fine with that because self harm. Well, AI aint fine with the detective's revolver, so..

If that's the case then the bolas aren't dumb. It's just that really small side effect to the shock animation. If that's the problem than I can throw it out the window, and keep it at a level where it only stuns.

> Journal wont be used. Just like the majority of the detective's gear in his office.

Well that can't really be decided by you right? It's a photo book where they can throw all of their photos into. And given that there's a labeler. Give it time and see what comes out of it. I still see Detective's running around with evidence bags, and actually securely storing evidence. It's the incentive that counts. If you want to motivate players to use a certain mechanic you have to give them something to carry it out. This is the carrier.

EMP Grenades. You can like them or hate them. They're five one-use grenades expected to be used for the same reason barrier grenades, a box of flashbangs, or teargas would ever be taken out for. In most rounds, they'll never get picked up.

**Comment by ghost** [comment]
It'd be better if they were just reinforced bola rather than instant GG throwingtraps where no matter how powerful the antag is if they get hit by one they are stunlocked and adrenals can't help them out because by the time they resist out they're tazed.

Since bola go through everything but shielded suits, even nukeops won't be safe.

Just make them regular/reinforced bola instead of chainstun bola. Otherwise, any antagonist hit by one is GG'd, where normally a stun resist ability might save your ass.

**Comment by Super3222** [comment]
Reinforced Bolas isn't that exciting tbh, Adam. I feel like stunning bolas is pretty appropriate given that it's security.

How about a threshold on the stun that will fill over the amount of time it takes to remove the actual bolas. So security officers can't be lazy, throw one at someone, and call it a day. They will actually have to track them. 

In that sense, if you get hit by a bola, after the shock you can still run away, and try to take it off without a getting any additional ones. I could even place a threshold/cooldown before the first shock even happens if it's really necessary.

**Comment by Alek2ander** [comment]
No, don't give security an ID-locked weapon. Seriously, this goes against the spirit of robustness.
If security has _that_ much problems dealing with stuff (which I don't see), give them... I don't know, ribbed gloves which prevent disarms from hand (but not item drop from the weaken outcome of a disarm).

**Comment by Super3222** [comment]
I don't know about ribbed gloves, but I can happily remove the ID restraint after I apply this first nerf.

**Comment by Super3222** [comment]
In the last commit:

Electrical Bolas have been reworked somewhat.
- They no longer start with 6 charges. Instead they start with 1.
- If one is tangled around your leg you have 10 seconds until it begins to shock you after every step you take.
- The charge number will still deplete like normal. Every step you take, there's a shock, and the amount of charges left determines how much you're stunned for and weakened.
- Electrical bolas are no longer ID restricted
- To gain more charges officers will need to "charge" the bolas by activating it in their hands.
- The charges will grow the longer it's in their hands for. The charge number cannot go over 6.
- There is a second timer which is used to kill the charges. This will only happen if an officer takes too long to actually use the bola. If this happens the charge rate will be reduced to 1.
- Bolas are no longer one use. To balance his, when a bola is successfully used it will take an additional 20 seconds (per use) to charge it.

**Comment by Cruix** [comment]
The bolas are a bad idea. Security is already insanely powerful against most antags with their ranged instant long-term stun, AOE flashbangs, and free agency to stun and search anyone for any reason at all, and instantly mark anyone who does not comply to be hunted down without question by every sec officer and robot on the station. There are very few counters to all of security's weapons - energy-shield wielding ops, dual eswords with sunglasses and earmuffs, or being high as fuck on stims or one of Bob Blitzer's powergame concoctions. In addition, most of these counters are extremely obvious to just about everyone.

Giving security these stun-bolas makes the situation even worse, since at least in the past you had a ghost of a chance trying to click every security officer on the station with your esword before they close to melee and click you once with their stunbaton. Now they can stand at comfortable range and hurl one-stun-and-you're-done bolas at you until you drop, or put away your esword to catch them and fall prey to the rest of their arsenal of fuck-you-antag weapons. It does not matter whether you have ten or fifty seconds before it starts shocking you, they will just follow you around at the edge of the screen until you try to resist out and have to stand still for however long it takes, or succumb and get shocked.

The EMP grenades are the same. EMPs are a niche mechanic and having the only easy source of them for security be a single stealable bulky weapon makes sense. If they really need EMP grenades, they can go ask chemistry or go on a grand adventure making some for themselves.

**Comment by Super3222** [comment]
Catching someone in a bola is the equivalent to catching someone in a taser bolt. The only difference is that the bola will stick to you, and punish you for moving.

EMP grenades, I'm not sure what you even mean.

**Comment by ShadowDeath6** [comment]
this seems like way too much powercreep for sec, and the bolas are an awful idea. Sec has way too many hardstuns already.

**Comment by Super3222** [comment]
> this seems like way too much powercreep for sec

I doubt it tbh. You've gotta play security a few rounds to actually know the beauty in having some of this equipment. But the bolas may have been excessive. I could've toned it down to 1 stun or something.

**Comment by Kmc2000** [comment]
Rename this to "removes security: THE PR!" and this will get accepted, no new features allowed

**Comment by Super3222** [comment]
wait let me port kmc's binocular spites. they're good

**Comment by ShadowDeath6** [comment]
If warden gets emp nades nerf flashbang effect on borgs :^)

**Comment by Super3222** [comment]
No.

**Comment by BOBsHERE** [comment]
The bola's are basically a hard counter to double Esword. Im not sure if this is a good thing for generic security to have access to. Also its a hard counter to any kind of stimulant. Also not sure if this is a good thing (maybe I'm biased here). Binoculars are an awesome idea. Sec should have a few EMP grenades (although that majorly fucks with traitor ai balance. I can kill an ai in literally five seconds with access to AOE EMP's because of the apc controlling the turrets. Just things to think about.

**Comment by ShadowDeath6** [comment]
Stimpacks actually ignore this unless legcuffs snowflake make you slow down no matter what

**Comment by Super3222** [comment]
Freedom implants instantly remove bolas, so there's a counter in that. Stimpacks as well, I presume.

File: code/modules/library/lib_items.dm [hunk 1]
```diff
@@ -313,4 +313,4 @@
 		user << "<font color=green>Computer has been associated with this unit.</font>"
 	else
 		user << "<font color=red>No associated computer found. Only local scans will function properly.</font>"
-	user << "\n"
+	user << "\n"
```
Comment (Cruix) [inline code/modules/library/lib_items.dm:316]: Wut
I don't know if I can merge this
Balance implications are huge

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -138,6 +139,19 @@
 	pictures_max = 30
 	pictures_left = 30
 
+/obj/item/weapon/storage/photobook/detective
+	name = "detective's journal"
+	desc = "October 12th, 1985. Tonight, a comedian died in New York. Somebody knows why."
+	can_hold = list(/obj/item/weapon/photo, /obj/item/weapon/hand_labeler, /obj/item/weapon/paper)
```
Comment (Cruix) [inline code/modules/paperwork/photography.dm]: Increase the max weight class so it can actually hold the hand labeler

Comment (Cruix) [reply code/modules/paperwork/photography.dm]: Also make it able to hold cameras
